drivers/platform/x86/hp-wmi.c

```diff
@@ -275,9 +287,13 @@ static int hp_wmi_als_state(void)
 static int hp_wmi_dock_state(void)
 {
 	int state = 0;
-	int ret = hp_wmi_perform_query(HPWMI_HARDWARE_QUERY, 0, &state,
-				       sizeof(state), sizeof(state));
+	int ret;
+
+	if (dmi_check_system(not_dockable))
+		return 0;
```

Comment:
You could add a global :)

Personally I think it would be worth adding a device-specific struct, but you know better than me what the upstream preference would be on this situation, so go ahead with what you would be easier to push upstream later.